### PR TITLE
[FLINK-6280] [scripts] Allow logging with Java flags

### PR DESCRIPTION
**Pull request by greghogan**
Evaluate user-defined Java options immediately before starting services and rotate all log files.

### CODE REVIEW

**Comment by StephanEwen** [comment]
Can you explain a bit more what this change does exactly? Does it make sure that if you put bash commands into the java opts to compute flags, it evaluates those?

Can you elaborate why this needs a change in the log rotation logic?

**Comment by greghogan** [comment]
I had provided some examples in the JIRA, but yes, it allows delaying evaluation of the java opts (with double quotes) until immediately before the daemon is started. This allows creating additional log files such as for JFR or JitWatch (the two use cases I have needed, I expect there are more). In addition the log rotation, Flink may be running multiple TaskManagers on the same instance so reusing Flink's log naming scheme prevents overwriting files.

**Comment by greghogan** [comment]
@StephanEwen added some documentation covering use and examples.

**Comment by StephanEwen** [comment]
Okay, got it. The docs and examples did help a lot.

This looks good to me, +1 to merge

File: docs/monitoring/application_profiling.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+specific language governing permissions and limitations
+under the License.
+-->
+
+* ToC
+{:toc}
+
+## Overview of Custom Logging with Apache Flink
+
+The standalone JobManager, TaskManager, HistoryServer, and ZooKeeper daemons by write `stdout` and `stderr` to files
```
Comment (zentol) [inline docs/monitoring/application_profiling.md]: the "by" between "daemons by write" should be removed.

**Comment by StephanEwen** [comment]
Feel free to merge...

**Pull request closed without merging**